src/cript/nodes/core.py

```diff
@@ -73,3 +90,90 @@ def json(self):
         from cript.nodes.util import NodeEncoder
 
         return json.dumps(self, cls=NodeEncoder)
+
+    def find_children(self, search_attr: dict, search_depth: int = -1) -> List:
+        """
+        Finds all the children in a given tree of nodes (specified by its root),
+        that match the criteria of search_attr.
+        If a node is present multiple times in the graph, it is only once in the search results.
+
+        search_dept: Max depth of the search into the tree. Helpful if circles are expected. -1 specifies no limit
+
+        search_attr: dict
+           Dictionary that specifies which JSON attributes have to be present in a given node.
+           If an attribute is a list, it it is suffiecient if the specified attributes are in the list,
+           if others are present too, that does not exclude the child.
+
+           Example: search_attr = `{"node": "Parameter"}` finds all "Parameter" nodes.
+                    search_attr = `{"node": "Algorithm", "parameter": {"name" : "update_frequency"}}`
+                                           finds all "Algorithm" nodes, that have a parameter "update_frequency".
+                                           Since parameter is a list an alternative notation is
+                                           ``{"node": "Algorithm", "parameter": [{"name" : "update_frequency"}]}`
+                                           and Algorithms are not excluded they have more paramters.
+                    search_attr = `{"node": "Algorithm", "parameter": [{"name" : "update_frequency"},
+                                           {"name" : "cutoff_distance"}]}`
+                                           finds all algoritms that have a parameter "update_frequency" and "cutoff_distance".
+
+        """
+
+        def is_attr_present(node: BaseNode, key, value):
```

Comment:
If this ends up being applicable code to other functions in this file I would take this out of the scope of just this function, but it looks fine for now. Additionally, I believe it doesn't necessarily need to be in the function in the first place other than to try to hide its existence from the public, because it doesn't use anything native to the function in the way it is written